client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/ExternalServiceInfo.java

```diff
@@ -0,0 +1,51 @@
+package org.evomaster.client.java.instrumentation;
+
+import java.io.Serializable;
+
+/**
+ * Information related to the external service calls
+ */
+public class ExternalServiceInfo implements Serializable {
+
+    /**
+     * Contains the remote hostname
+     */
+    private final String remoteHostname;
+
+    private final String protocol;
+
+    private final int remotePort;
+
+    private final String mockHostname;
+
+    private final int mockHostPort;
+
+    public ExternalServiceInfo(String protocol, String remoteHostname, int remotePort, String mockHostname, int mockHostPort) {
```

Comment:
as u are using this class inside a `Set`, then must define `equals/hashcode` methods. IDE can generate those for you